CMakeLists.txt

```diff
@@ -1079,9 +1079,6 @@ check_symbol_exists(_strtoi64      "${CURL_INCLUDES}" HAVE__STRTOI64)
 check_symbol_exists(strerror_r     "${CURL_INCLUDES}" HAVE_STRERROR_R)
 check_symbol_exists(siginterrupt   "${CURL_INCLUDES}" HAVE_SIGINTERRUPT)
 check_symbol_exists(getaddrinfo    "${CURL_INCLUDES}" HAVE_GETADDRINFO)
-if(NOT HAVE_GETADDRINFO)
-  set(HAVE_GETADDRINFO_THREADSAFE OFF)
-endif()
```

Comment:
wouldn't it make sense to leave this

Reply:
It wouldn't hurt, but why guard against this twice?

Also without it, our CI wouldn't be really verifying this PR.

Reply:
I don't see where else it is checked. What I am saying is it wouldn't make sense that HAVE_GETADDRINFO_THREADSAFE is defined when not HAVE_GETADDRINFO.

Reply:
Ok. IMO it wouldn't hurt to leave it there.

Reply:
Fine with me, I'll put it back there.

Reply:
@jay: Rephrasing the Windows logic as this:
```cmake
if(WIN32)
   set(HAVE_GETADDRINFO_THREADSAFE ${HAVE_GETADDRINFO})
 endif()
```
would IMO express our intent much cleaner and self contained in these three lines, with identical results. It would also allow dropping the related line from `WindowsCache.cmake`.

Meaning: When we have `getaddrinfo()`, it is always threadsafe (on Windows).

What do you think?

Reply:
Works for me